File: myapp/lib/models/product.dart
import 'package:meta/meta.dart';

class Product {
  final int id;
  final String name;
  final String url;
  final int year;
  final double price;
  final quantity;

  Product({
    @required this.id,
    @required this.name,
    this.url,
    @required this.year,
    @required this.price,
    @required this.quantity}):
        assert(name != null),
        assert(year != null),
        assert(quantity >= 0)
  ;
  factory Product.fromJSON({Map<String, dynamic> data}) {
    return Product(
      id: data['id'] ?? 0,
      name: data['name'] ?? '',
      url: data['url'] ?? '',
      year: data['year'] ?? 200,
      quantity: data['role'] ?? 1,
    );
  }

}

File: nodejsBackend/models/product.js
const { DataTypes, Sequelize } = require('sequelize');
//sequelize-auto -o "./models1" -h localhost -u root -p 3306 -x "" -e mysql -d NodejsBackend
module.exports = async (sequelize) => {	
	return sequelize.define('Product', {
        id: {
          autoIncrement: true,
          type: DataTypes.INTEGER(11),
          allowNull: false,
          primaryKey: true
        },
        name: {
          type: DataTypes.STRING(400),
          allowNull: false,
          unique: true
        },
        url: {
          type: DataTypes.STRING(400),
          allowNull: true
        },
        year: {
          type: DataTypes.INTEGER(11),
          allowNull: true,
          defaultValue: 2000
        },
        price: {
          type: DataTypes.FLOAT,
          allowNull: true,
          defaultValue: 0
        },
        quantity: {
          type: DataTypes.INTEGER(11),
          allowNull: true,
          defaultValue: 0
        }
      }, {
        sequelize,
        tableName: 'Product',
        timestamps: false,
      });
}

